--- ./dwm.c.orig	2021-01-04 11:16:05.649734000 +0100
+++ dwm.c	2021-01-04 11:21:39.903724000 +0100
@@ -268,6 +268,7 @@
 static int xerrordummy(Display *dpy, XErrorEvent *ee);
 static int xerrorstart(Display *dpy, XErrorEvent *ee);
 static void zoom(const Arg *arg);
+static void switchstext(const Arg *arg);
 
 /* variables */
 static Systray *systray =  NULL;
@@ -304,6 +305,7 @@
 static Drw *drw;
 static Monitor *mons, *selmon;
 static Window root, wmcheckwin;
+static int stexton = 1;
 
 /* configuration, allows nested code to access above variables */
 #include "config.h"
@@ -2248,13 +2250,23 @@
 void
 updatestatus(void)
 {
-	if (!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
-		strcpy(stext, "dwm-"VERSION);
+	if (stexton) {
+		if (!gettextprop(root, XA_WM_NAME, stext, sizeof(stext)))
+			strcpy(stext, "dwm-"VERSION);
+	} else
+		strcpy(stext, "");
 	drawbar(selmon);
 	updatesystray();
 }
 
 void
+switchstext(const Arg *arg)
+{
+	stexton ^= 1;
+	updatestatus();
+}
+
+void
 updatesystrayicongeom(Client *i, int w, int h)
 {
 	if (i) {
