--- dwm.c.orig	Sat Dec 12 13:53:31 2020
+++ dwm.c	Sun Dec 13 09:22:21 2020
@@ -79,7 +79,7 @@
 enum { NetSupported, NetWMName, NetWMState, NetWMCheck,
        NetSystemTray, NetSystemTrayOP, NetSystemTrayOrientation, NetSystemTrayOrientationHorz,
        NetWMFullscreen, NetActiveWindow, NetWMWindowType,
-       NetWMWindowTypeDialog, NetClientList, NetLast }; /* EWMH atoms */
+       NetWMWindowTypeDialog, NetWMDesktop, NetClientList, NetLast }; /* EWMH atoms */
 enum { Manager, Xembed, XembedInfo, XLast }; /* Xembed atoms */
 enum { WMProtocols, WMDelete, WMState, WMTakeFocus, WMLast }; /* default atoms */
 enum { ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle,
@@ -270,6 +270,7 @@
 static void zoom(const Arg *arg);
 static void bstack(Monitor *m);
 static void switchstextvis(const Arg *arg);
+static void updatewmdesktop(Client *c);
 
 
 /* variables */
@@ -1170,6 +1171,7 @@
 		c->mon = selmon;
 		applyrules(c);
 	}
+	updatewmdesktop(c);
 
 	if (c->x + WIDTH(c) > c->mon->mx + c->mon->mw)
 		c->x = c->mon->mx + c->mon->mw - WIDTH(c);
@@ -1639,6 +1641,7 @@
 	detachstack(c);
 	c->mon = m;
 	c->tags = m->tagset[m->seltags]; /* assign tags of target monitor */
+	updatewmdesktop(c);
 	attach(c);
 	attachstack(c);
 	focus(NULL);
@@ -1797,6 +1800,7 @@
 	netatom[NetWMFullscreen] = XInternAtom(dpy, "_NET_WM_STATE_FULLSCREEN", False);
 	netatom[NetWMWindowType] = XInternAtom(dpy, "_NET_WM_WINDOW_TYPE", False);
 	netatom[NetWMWindowTypeDialog] = XInternAtom(dpy, "_NET_WM_WINDOW_TYPE_DIALOG", False);
+	netatom[NetWMDesktop] = XInternAtom(dpy, "_NET_WM_DESKTOP", False);
 	netatom[NetClientList] = XInternAtom(dpy, "_NET_CLIENT_LIST", False);
 	xatom[Manager] = XInternAtom(dpy, "MANAGER", False);
 	xatom[Xembed] = XInternAtom(dpy, "_XEMBED", False);
@@ -1898,6 +1902,7 @@
 {
 	if (selmon->sel && arg->ui & TAGMASK) {
 		selmon->sel->tags = arg->ui & TAGMASK;
+		updatewmdesktop(selmon->sel);
 		focus(NULL);
 		arrange(selmon);
 	}
@@ -1981,6 +1986,7 @@
 	newtags = selmon->sel->tags ^ (arg->ui & TAGMASK);
 	if (newtags) {
 		selmon->sel->tags = newtags;
+		updatewmdesktop(selmon->sel);
 		focus(NULL);
 		arrange(selmon);
 	}
@@ -2569,6 +2575,14 @@
 				tx += WIDTH(c);
 		}
 	}
+}
+
+void
+updatewmdesktop(Client *c)
+{
+	unsigned int data[] = { c->tags };
+	XChangeProperty(dpy, c->win, netatom[NetWMDesktop], XA_CARDINAL, 32,
+		PropModeReplace, (unsigned char*)data, 1);
 }
 
 int
