--- ./dwm.c.orig	2021-01-04 11:23:08.203057000 +0100
+++ dwm.c	2021-01-04 11:27:35.651934000 +0100
@@ -79,7 +79,7 @@
 enum { NetSupported, NetWMName, NetWMState, NetWMCheck,
        NetSystemTray, NetSystemTrayOP, NetSystemTrayOrientation, NetSystemTrayOrientationHorz,
        NetWMFullscreen, NetActiveWindow, NetWMWindowType,
-       NetWMWindowTypeDialog, NetClientList, NetLast }; /* EWMH atoms */
+       NetWMWindowTypeDialog, NetWMDesktop, NetClientList, NetLast }; /* EWMH atoms */
 enum { Manager, Xembed, XembedInfo, XLast }; /* Xembed atoms */
 enum { WMProtocols, WMDelete, WMState, WMTakeFocus, WMLast }; /* default atoms */
 enum { ClkTagBar, ClkLtSymbol, ClkStatusText, ClkWinTitle,
@@ -269,6 +269,7 @@
 static int xerrorstart(Display *dpy, XErrorEvent *ee);
 static void zoom(const Arg *arg);
 static void switchstext(const Arg *arg);
+static void updatewmdesktop(Client *c);
 
 /* variables */
 static Systray *systray =  NULL;
@@ -1169,6 +1170,7 @@
 		c->mon = selmon;
 		applyrules(c);
 	}
+	updatewmdesktop(c);
 
 	if (c->x + WIDTH(c) > c->mon->mx + c->mon->mw)
 		c->x = c->mon->mx + c->mon->mw - WIDTH(c);
@@ -1638,6 +1640,7 @@
 	detachstack(c);
 	c->mon = m;
 	c->tags = m->tagset[m->seltags]; /* assign tags of target monitor */
+	updatewmdesktop(c);
 	attach(c);
 	attachstack(c);
 	focus(NULL);
@@ -1796,6 +1799,7 @@
 	netatom[NetWMFullscreen] = XInternAtom(dpy, "_NET_WM_STATE_FULLSCREEN", False);
 	netatom[NetWMWindowType] = XInternAtom(dpy, "_NET_WM_WINDOW_TYPE", False);
 	netatom[NetWMWindowTypeDialog] = XInternAtom(dpy, "_NET_WM_WINDOW_TYPE_DIALOG", False);
+	netatom[NetWMDesktop] = XInternAtom(dpy, "_NET_WM_DESKTOP", False);
 	netatom[NetClientList] = XInternAtom(dpy, "_NET_CLIENT_LIST", False);
 	xatom[Manager] = XInternAtom(dpy, "MANAGER", False);
 	xatom[Xembed] = XInternAtom(dpy, "_XEMBED", False);
@@ -1897,6 +1901,7 @@
 {
 	if (selmon->sel && arg->ui & TAGMASK) {
 		selmon->sel->tags = arg->ui & TAGMASK;
+		updatewmdesktop(selmon->sel);
 		focus(NULL);
 		arrange(selmon);
 	}
@@ -1980,6 +1985,7 @@
 	newtags = selmon->sel->tags ^ (arg->ui & TAGMASK);
 	if (newtags) {
 		selmon->sel->tags = newtags;
+		updatewmdesktop(selmon->sel);
 		focus(NULL);
 		arrange(selmon);
 	}
@@ -2538,6 +2544,14 @@
 	pop(c);
 }
 
+void
+updatewmdesktop(Client *c)
+{
+	unsigned int data[] = { c->tags };
+	XChangeProperty(dpy, c->win, netatom[NetWMDesktop], XA_CARDINAL, 32,
+		PropModeReplace, (unsigned char*)data, 1);
+}
+
 int
 main(int argc, char *argv[])
 {
