--- ./dwm.c.orig	2021-01-04 11:28:22.799990000 +0100
+++ dwm.c	2021-01-04 11:31:50.752667000 +0100
@@ -49,12 +49,13 @@
 #define CLEANMASK(mask)         (mask & ~(numlockmask|LockMask) & (ShiftMask|ControlMask|Mod1Mask|Mod2Mask|Mod3Mask|Mod4Mask|Mod5Mask))
 #define INTERSECT(x,y,w,h,m)    (MAX(0, MIN((x)+(w),(m)->wx+(m)->ww) - MAX((x),(m)->wx)) \
                                * MAX(0, MIN((y)+(h),(m)->wy+(m)->wh) - MAX((y),(m)->wy)))
-#define ISVISIBLE(C)            ((C->tags & C->mon->tagset[C->mon->seltags]))
 #define LENGTH(X)               (sizeof X / sizeof X[0])
 #define MOUSEMASK               (BUTTONMASK|PointerMotionMask)
 #define WIDTH(X)                ((X)->w + 2 * (X)->bw)
 #define HEIGHT(X)               ((X)->h + 2 * (X)->bw)
 #define TAGMASK                 ((1 << LENGTH(tags)) - 1)
+#define HIDDENTAG               (TAGMASK + 1)
+#define ISVISIBLE(C)            (!(C->tags & HIDDENTAG) && (C->tags & C->mon->tagset[C->mon->seltags]))
 #define TEXTW(X)                (drw_fontset_getwidth(drw, (X)) + lrpad)
 
 #define SYSTEM_TRAY_REQUEST_DOCK    0
@@ -270,6 +271,7 @@
 static void zoom(const Arg *arg);
 static void switchstext(const Arg *arg);
 static void updatewmdesktop(Client *c);
+static void hide(const Arg *arg);
 
 /* variables */
 static Systray *systray =  NULL;
@@ -618,6 +620,11 @@
 			setfullscreen(c, (cme->data.l[0] == 1 /* _NET_WM_STATE_ADD    */
 				|| (cme->data.l[0] == 2 /* _NET_WM_STATE_TOGGLE */ && !c->isfullscreen)));
 	} else if (cme->message_type == netatom[NetActiveWindow]) {
+		if (c->tags & HIDDENTAG) {
+			setclientstate(c, NormalState);
+			c->tags &= ~HIDDENTAG;
+			arrange(c->mon);
+		}
 		for (i = 0; i < LENGTH(tags) && !((1 << i) & c->tags); i++);
 		if (i < LENGTH(tags)) {
 			selmon = c->mon;
@@ -2552,6 +2559,15 @@
 		PropModeReplace, (unsigned char*)data, 1);
 }
 
+void
+hide(const Arg *arg)
+{
+	selmon->sel->tags |= HIDDENTAG;
+	setclientstate(selmon->sel, IconicState);
+	focus(NULL);
+	arrange(selmon);
+}
+
 int
 main(int argc, char *argv[])
 {
